### DISCUSSION TITLE
Routing files - Can I somehow use folders instead of files?

### DISCUSSION THREAD

**Discussion by michaelschufi** [vercel/next.js#53379] [App Router]
I.e. instead of `pages.tsx` or `layout.tsx`, `pages/index.tsx` or `layout/index.tsx`?

For example, I have multiple subcomponents used in my root layout. Since the layout is the only place those components get used, I place them into a folder. The issue is, they are completely separate in the file tree. This kind of defeats the purpose a bit of having all your files in the same place in a file-based router.

What I currently need to do
```
- app
  - layout
    - index.tsx <-- actual layout component that uses files below.
    - home-button.tsx
    - nav-buttons.tsx
  - route-a
    - page.tsx
  - route-b
    - page.tsx
  - route-c ...
  - layout.tsx <-- is separate to the actual layout folder.
```

What I would like to do:
```
- app
  - layout
    - index.tsx <-- actual layout component that uses files below.
    - home-button.tsx
    - nav-buttons.tsx
  - route-a
    - page.tsx
  - route-b
    - page.tsx
  - route-c ...
  <-- No layout.tsx required. `layout/index.tsx` gets found by Next.js
```

I have a similar issue with the page. However, there I can, instead of

```
- app
  - route-a
    - home-button.tsx
    - route-a-sub
      ....
    - route-b-sub
    - nav-buttons.tsx
    - page.tsx
```

this
```
- app
  - route-a
    - (page)
      - page.tsx
      - home-button.tsx
      - nav-buttons.tsx
```

**Comment by arvinxx** [comment]
really need the `layout/index.tsx` for better organization. currently we have a `_layout` to organize the layout files :

```
.
├── (mobile)
├── @nav
├── _layout
│   ├── Desktop.tsx
│   ├── Mobile.tsx
│   └── type.ts
├── layout.tsx

```


if we can put `layout.tsx` and `_layout` folder into a `layout` folder, it will be more clean! 


```
.
├── (mobile)
├── @nav
├── layout
│   ├── Desktop.tsx
│   ├── Mobile.tsx
│   ├── type.ts
│   └── index.tsx
```
